src/Workspaces/Core/Portable/FindSymbols/SymbolFinder.FindReferencesServerCallback.cs

```diff
@@ -65,39 +65,37 @@ public async ValueTask OnDefinitionFoundAsync(SerializableSymbolGroup dehydrated
             await progress.OnDefinitionFoundAsync(symbolGroup, cancellationToken).ConfigureAwait(false);
         }
 
-        public ValueTask OnReferencesFoundAsync(
+        public async ValueTask OnReferencesFoundAsync(
             ImmutableArray<(SerializableSymbolGroup serializableSymbolGroup, SerializableSymbolAndProjectId serializableSymbol, SerializableReferenceLocation reference)> references,
             CancellationToken cancellationToken)
         {
-            return progress.OnReferencesFoundAsync(ConvertAsync(cancellationToken), cancellationToken);
+            using var _ = ArrayBuilder<(SymbolGroup group, ISymbol symbol, ReferenceLocation location)>.GetInstance(references.Length, out var rehydrated);
```

Comment:
>ArrayBuilder [](http://example.com/codeflow?start=26&length=12)

What's the point of using an arraybuilder here, won't there only ever be a single item added to it before it gets cleared?

Reply:
fixed!